compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1972,4 +1972,36 @@ object messages {
     val msg = hl"Cannot extend ${"sealed"} $pclazz in a different source file"
     val explanation = "A sealed class or trait can only be extended in the same file as its declaration"
   }
+
+  case class UnableToEmitSwitch()(implicit ctx: Context)
+  extends Message(UnableToEmitSwitchID) {
+    val kind = "Syntax"
+    val msg = hl"""Could not emit switch for ${"@switch"} annotated match"""
+    val explanation = {
+      val errorCodeExample =
+        """val middle = 'm'
+          |val last   = 'z'
+          |
+          |val number   = (middle: @switch) match {
+          |  case 'a'  => 1
+          |  case 'm'  => 13
+          |  case last => 26  //a non-literal may prevent switch generation: this would not compile.
```

Comment:
`case _` is unreachable. You might want to either replace
```
case last => ...
```
with
```
case `last` => ...
```
or rename `last` to `Last`

Reply:
Fixed.